services/s3/src/test/java/software/amazon/awssdk/services/s3/S3PresignerTest.java

```diff
@@ -325,4 +330,94 @@ public void putObject_Sigv4PresignerHonorsSignatureDuration() {
+                .build())
+                .build();
+
+        PresignedGetObjectRequest presignedRequest =
+                presigner.presignGetObject(r -> r.signatureDuration(Duration.ofMinutes(5))
+                        .getObjectRequest(go -> go.bucket(BUCKET)
+                                .key("bar")));
+
+        assertThat(presignedRequest.httpRequest().host()).isEqualTo(String.format("%s.s3-accelerate.amazonaws.com", BUCKET));
+    }
+
+    /**
+     * Dualstack uses regional endpoints that support virtual addressing.
+     */
+    @Test
+    public void dualstackEnabled_UsesVirtualAddressingWithDualstackEndpoint() throws Exception {
+        S3Presigner presigner = presignerBuilder().serviceConfiguration(S3Configuration.builder()
+                .dualstackEnabled(true)
+                .build())
+                .build();
+
+        PresignedGetObjectRequest presignedRequest =
+                presigner.presignGetObject(r -> r.signatureDuration(Duration.ofMinutes(5))
+                        .getObjectRequest(go -> go.bucket(BUCKET)
+                                .key("bar")));
+
+        assertThat(presignedRequest.httpRequest().host()).contains(String.format("%s.s3.dualstack.us-west-2.amazonaws.com", BUCKET));
+    }
+
+    /**
+     * Dualstack also supports path style endpoints just like the normal endpoints.
+     */
+    @Test
+    public void dualstackAndPathStyleEnabled_UsesPathStyleAddressingWithDualstackEndpoint() throws Exception {
+        S3Presigner presigner = presignerBuilder().serviceConfiguration(S3Configuration.builder()
+                .dualstackEnabled(true)
+                .pathStyleAccessEnabled(true)
+                .build())
+                .build();
+
+        PresignedGetObjectRequest presignedRequest =
+                presigner.presignGetObject(r -> r.signatureDuration(Duration.ofMinutes(5))
+                        .getObjectRequest(go -> go.bucket(BUCKET)
+                                .key("bar")));
+
+        assertThat(presignedRequest.httpRequest().host()).isEqualTo("s3.dualstack.us-west-2.amazonaws.com");
+    }
+
+    /**
+     * When dualstack and accelerate are both enabled there is a special, global dualstack endpoint we must use.
+     */
+    @Test
+    public void dualstackAndAccelerateEnabled_UsesDualstackAccelerateEndpoint() throws Exception {
+        S3Presigner presigner = presignerBuilder().serviceConfiguration(S3Configuration.builder()
+                .dualstackEnabled(true)
+                .accelerateModeEnabled(true)
+                .build())
+                .build();
+
+        PresignedGetObjectRequest presignedRequest =
+                presigner.presignGetObject(r -> r.signatureDuration(Duration.ofMinutes(5))
+                        .getObjectRequest(go -> go.bucket(BUCKET)
+                                .key("bar")));
+
+        assertThat(presignedRequest.httpRequest().host()).isEqualTo(String.format("%s.s3-accelerate.dualstack.amazonaws.com", BUCKET));
```

Comment:
Nice! Thanks for testing this. 

For the other features:
1. pathStyleAccessEnabled, useArnRegionEnabled both affect the endpoint. If they're working, can we add a test for them? If they're not working, maybe we should throw an exception if they're configured.
2. chunkedEncodingEnabled and checksumValidationEnabled don't make sense for presigned requests right now. Can we add some javadoc to mention that they currently aren't used for presigning?